engines/private/private.cpp

```diff
@@ -120,40 +121,66 @@ void PrivateEngine::initializePath(const Common::FSNode &gamePath) {
 	SearchMan.addDirectory(gamePath.getPath(), gamePath, 0, 10);
 }
 
-Common::Error PrivateEngine::run() {
+Common::SeekableReadStream *PrivateEngine::loadAssets() {
 
-	assert(_installerArchive.open("SUPPORT/ASSETS.Z"));
+	Common::File *test = new Common::File();
 	Common::SeekableReadStream *file = NULL;
-	// if the full game is used
-	if (!isDemo()) {
-		assert(_installerArchive.hasFile("GAME.DAT"));
-		file = _installerArchive.createReadStreamForMember("GAME.DAT");
+
+	if (isDemo() && test->open("SUPPORT/ASSETS/DEMOGAME.WIN"))
+		file = (Common::SeekableReadStream *)test;
```

Comment:
Why this cast? I think, simple `file = test` will work here as well

Reply:
Explicit casting makes me feel good!, but you are right. No need for that. Fixed!